### PR TITLE
Enforce the use of C++11

### PR DESCRIPTION
**Pull request by Nicogene**
This PR enforce the use of C++11, both for using and compiling `RTF`.
`NULL` substituted by `nullptr`, `tinythread by std::thread`.

Please review code.

### CODE REVIEW

File: src/plugins/dll/src/SharedLibrary.cpp [hunk 1]
```diff
@@ -38,54 +38,54 @@ bool SharedLibrary::open(const char *filename) {
...
+    LPTSTR msg = nullptr;
     FormatMessage(
        FORMAT_MESSAGE_FROM_SYSTEM |FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_IGNORE_INSERTS,
-       NULL, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
-       (LPTSTR)&msg, 0, NULL);
+       nullptr, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+       (LPTSTR)&msg, 0, nullptr);
 
-    if(msg != NULL) {
+    if(msg != nullptr) {
```
Comment (traversaro) [inline src/plugins/dll/src/SharedLibrary.cpp]: I would change this checks in `if(msg)` and the negative check `if(msg == nullptr)` in `if(!msg)`.

Comment (Nicogene) [reply src/plugins/dll/src/SharedLibrary.cpp]: I think that it is better to move the change you requested in a separated PR, we have to do some cleanups (e.g invert `RTF_ASSERT_IF` logic) and we can add also these changes for the nullptr checks.

Comment (traversaro) [reply src/plugins/dll/src/SharedLibrary.cpp]: Ok!

**Comment by aerydna** [comment]
merge thanks!

**Pull request merged**